tools/flowc/manipulation/tail_call.flow

```diff
@@ -151,142 +142,205 @@ fcReturnExps(e : FiExp, acc : [FiExp]) -> [FiExp] {
     }
 }
 
+fcVarExistsInNonReturnExps(ex : FiExp, v : string, non_return: bool) -> bool {
+    switch (ex) {
+		FiLet(__, __,e1,e2,__, __): {
+			fcVarExistsInNonReturnExps(e1, v, true) ||
+			fcVarExistsInNonReturnExps(e2, v, non_return);
+		}
+		FiIf(e1, e2, e3,__,__): {
+			fcVarExistsInNonReturnExps(e1, v, true) ||
+			fcVarExistsInNonReturnExps(e2, v, non_return) ||
+			fcVarExistsInNonReturnExps(e3, v, non_return);
+		}
+		FiSwitch(x,__,cs,__,__): {
+			(non_return && x.name == v) || exists(cs,\c -> fcVarExistsInNonReturnExps(c.body, v, non_return));
+		}
+		FiSeq(es,__,__): {
+			if (length(es) <= 1) false else
+			exists(take(es, length(es) - 1), \x -> fcVarExistsInNonReturnExps(x, v, true));
+		}
+		FiCallPrim(__,es,__,__): {
+			exists(es,\x -> fcVarExistsInNonReturnExps(x, v, non_return))
```

Comment:
true? instead of non_return

arr[index] = FiCallPrim
arr = es
-> ~=
getArray()[index]
-> ~sequence